geode-core/src/main/java/org/apache/geode/internal/cache/xmlcache/CacheCreation.java

```diff
@@ -1075,7 +1078,12 @@ public CancelCriterion getCancelCriterion() {
   }
 
   @Override
-  public InternalQueryService getQueryService() {
+  public QueryService getQueryService() {
+    return queryService;
+  }
+
+  @Override
+  public InternalQueryService getInternalQueryService() {
```

Comment:
Explanation: I changed impl methods that narrow the types (return or params)  to match the interface. I added new methods with narrowed types where applicable (such as here).